client/src/main/java/org/apache/falcon/client/FalconClient.java

```diff
@@ -817,6 +823,15 @@ public ResourceBuilder addQueryParam(String paramName, List<LifeCycle> lifeCycle
             return this;
         }
 
+        public ResourceBuilder addQueryParam(String paramName, List<Entity> entities) {
+            if (entities != null && !entities.isEmpty()) {
+                for (Entity entity : entities) {
+                    resource = resource.queryParam(paramName, entity.toString());
```

Comment:
Isn't the query param being overwritten as it is using the same paramName for every entity?